plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/triggers/get_query_results/trigger.py

```diff
@@ -0,0 +1,40 @@
+import insightconnect_plugin_runtime
+import time
+from .schema import GetQueryResultsInput, GetQueryResultsOutput, Input, Output, Component
+
+# Custom imports below
+from ...util.util import Util
+
+
+class GetQueryResults(insightconnect_plugin_runtime.Trigger):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="get_query_results",
+            description=Component.DESCRIPTION,
+            input=GetQueryResultsInput(),
+            output=GetQueryResultsOutput(),
+        )
+
+    def run(self, params={}):
+        query = params.get(Input.QUERY)
+        tenants = params.get(Input.TENANTS)
+        limit = params.get(Input.LIMIT)
+        interval = abs(params.get(Input.FREQUENCY, 5))
+
+        end_time = Util.now_ms() - interval * 1000
+
+        self.logger.info("Initializing Get Query Results trigger for the Palo Alto Cortex XDR plugin.")
+
+        while True:
+            start_time = Util.now_ms()
+            execution_id = self.connection.xdr_api.start_xql_query(query, tenants, end_time, start_time)
+            query_results = insightconnect_plugin_runtime.helper.clean(
```

Comment:
This method does not clean nested dictionaries well. Do results contain nested dicts?

Reply:
I think there is one `dict` in it, but nothing much deeper 😃